workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/UnityComponentDetails.cs

```diff
@@ -26,11 +27,33 @@ public UnityComponentDetails(string package, ComponentDefinition componentDefini
 
             CommandDetails = componentDefinitionRaw.Commands
                 .Select(command => new UnityCommandDetails(command))
+                .Where(commandDetail =>
+                {
+                    // Return true to keep commands that do not have a name clash with the component
+                    if (!commandDetail.CommandName.Equals(ComponentName))
+                    {
+                        return true;
+                    }
+
+                    Console.Error.WriteLine($"Error in component \"{ComponentName}\". Command \"{commandDetail.RawCommandName}\" clashes with component name.");
```

Comment:
This should FAIL codegen. It shouldn't be possible to write schema that throws errors but still generates code.

Reply:
resolved offline